### PR TITLE
fix: hide chat button when user has no tickets

### LINKED ISSUE #8438: Chat icon visible even if user has not joined event
The chat icon of an event is visible even a user has not joined an event. Expected: Only user who have a ticket and who have joined an event (through the blue join event button) should see the chat icon. This means only when the sidepanel is activated should the chat be available.

Screenshot from 2022-03-17 14-33-35

### PR DESCRIPTION
**Pull request by progmatic-99**
Fixes #8438 

Screenshot 2022-03-18 at 10-58-05 FOSSASIA Summit 2022 eventyay

### CODE REVIEW

**Comment by mariobehling** [comment]
Goal is hide Chat button when user has no ticket and neither has joined the event.

**Comment by progmatic-99** [comment]
@mariobehling Yes, that is implemented in this PR.

**Review by pc-beast — COMMENTED** [review]
Chat icon doesn't shows up even when side panel is active.
image

**Comment by progmatic-99** [comment]
@pc-beast You bought a ticket or registered for the event??

**Comment by pc-beast** [comment]
Yes

File: app/components/public/stream/side-panel.hbs [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <button {{did-insert this.setup}} class="ui icon button stream-side-panel-button" {{action (mut this.shown) true}}>
   <i class="angle right icon"></i>
 </button>
+<div class="ui chat-stream-side-panel-button right">
+  <i class="rocketchat large icon p-4 chat-panel-icon" {{on "click" @showChatPanel}} role="button"></i>
+</div>
 {{#if this.shown}}
   <div class="d-flex flex-column stream-side-panel" >
```
Comment (pc-beast) [inline app/components/public/stream/side-panel.hbs:8]: chat icon shows up even if the chat is not enabled for event

**Review by pc-beast — APPROVED** [review]

**Pull request merged**